src/prefect/infrastructure/provisioners/container_instance.py

```diff
@@ -524,7 +525,6 @@ def _generate_acr_name(base_name: str):
     async def _get_or_create_registry(
         self,
         registry_name: str,
-        resource_group_name: str,
```

Comment:
Not something that needs to be addressed in this PR, but I think we need to be consistent about when we pass values in and when we get values from the class state. I'd lean more towards passing values in because I think it will make it easier to decompose this class in the future.

Reply:
Okay, I've kept that as is